docs/handbooks/language-implementation.md

```diff
@@ -0,0 +1,144 @@
+# Language Implementation Handbook
+
+## Foreword
+
+Implementing a new language in *jsii* is not just a matter of implementing code
+generation: mapping the *[jsii type system]* to a new programming language takes
+some amounts of API design, finding ways to represent APIs in the most idiomatic
+possible way. This is a craft that often requires trial and error, and the best
+(if not only) way to validate a proposal is to put it in front of users and
+seek feedback. As a consequence, this endeavor should be expected to span
+months, not weeks.
+
+
+## Scoping & Planning
+
+The first step of most successful projects is to start by scoping work out and
+establishing a baseline plan to execute on. For contributors not yet familiar
+with *jsii*, the [specification] document is a great place to start. In
+particular, the [New Language Intake] document provides a high-level view of the
+recommended process for implementing new language support.
+
+### Language Proposition RFC
+
+The recommended way to formalize the initial plan is to write it into an RFC
+hosted in the [CDK RFC repository]. Enough time has to be spent considering the
+requirements in order to get the work scoped and planned well, ensuring smooth
+execution.
+
+An additional benefit of following the RFC process is that it makes it easier
+to track learnings accumulated through the implementation process, as those will
+be tracked as comments or iterations on the RFC document.
+
+It is possible (and sometimes desirable) to start prototyping code-generation
+for the new language, as this can hlighlight implementation challenges that need
+to be discussed in the RFC document. In any case, examples of the API signatures
+that are expected to be rendered allow early feedback to be provided by possible
+future users, and still helps identify challenges.
+
+The following questions should be answered as early as possible in the process,
+to avoid surprises later on that result in significant re-engineering effort:
+
+* What does the generated APIs look like, for the typical API idioms?
+  - *Classes* (constructors, properties, methods, inheritance strategy, abstract
+    members, ...)
+    + Construction experience when last argument of constructor is a struct
+  - *Enums*
+  - *Interfaces* and *Structs* (properties, methods, inheritance strategy,
+    implementation, ...). In particular, how are new optional properties handled
+    (those are not considered breaking change within the [jsii type system]).
+  - *Structs* (properties, inheritance strategy, implementation, ...)
+* What information is needed in order for the code-generator to produce
+  artifacts? What should the configuration block look like?
```

Comment:
What's a configuration block? (half rhetorical to point out that as a reader I'm not yet familiar with this. the other half because I genuinely don't know 😉)

Reply:
Added a pitch to this earlier in the document.